suseconnect/migration.go

```diff
@@ -0,0 +1,562 @@
+package main
+
+// TODO LIST
+// * zypp_backup/zypp_restore functions
+// * zypper dup wrapper with options (these are mostly pass-through from plugin args)
+// *   passthrough zypper dup options:
+// *     --allow-vendor-change
+// *     --from
+// *     --repo
+// *     --debug-solver
+// *     --recommends
+// *     --no-recommends
+// *     --replacefiles
+// *     --details
+// *     --download (including --download-only)
+// * --selfupdate option
+// * --query plugin option
+// * --break-my-system option
+// * --product option (offline migration)
+// * obsolete repo disabling (including --disable-repos option)
+// * interactive migration mode
+// * snapshots (snapper wrapper)
+// * Leap -> SLES migration case
+
+import (
+	_ "embed"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"sort"
+	"syscall"
+
+	"github.com/SUSE/connect-ng/internal/connect"
+)
+
+var (
+	//go:embed migrationUsage.txt
+	migrationUsageText string
+	// flag indicating interuption by INT/TERM signal
+	interrupted bool
+
+	// ErrInterrupted is returned when execution was interrupted by INT/TERM signal
+	ErrInterrupted = errors.New("Interrupted")
+)
+
+func init() {
+}
+
+func migrationMain() {
+	var (
+		dummy             bool
+		verbose           bool
+		quiet             bool
+		nonInteractive    bool
+		autoAgreeLicenses bool
+		noSnapshots       bool
+		noSelfUpdate      bool
+		migrationNum      int
+		fsRoot            string
+	)
+
+	flag.Usage = func() {
+		fmt.Print(migrationUsageText)
+	}
+	// dummy flag to keep default but accept cli arg
+	flag.BoolVar(&dummy, "no-verbose", false, "")
+	flag.BoolVar(&verbose, "verbose", false, "")
+	flag.BoolVar(&verbose, "v", false, "")
+	// dummy flag to keep default but accept cli arg
+	flag.BoolVar(&dummy, "no-quiet", false, "")
+	flag.BoolVar(&quiet, "quiet", false, "")
+	flag.BoolVar(&quiet, "q", false, "")
+	flag.BoolVar(&nonInteractive, "non-interactive", false, "")
+	flag.BoolVar(&nonInteractive, "n", false, "")
+	flag.BoolVar(&autoAgreeLicenses, "auto-agree-with-licenses", false, "")
+	flag.BoolVar(&autoAgreeLicenses, "l", false, "")
+	flag.BoolVar(&noSnapshots, "no-snapshots", false, "")
+	// dummy flag to keep default but accept cli arg
+	flag.BoolVar(&dummy, "selfupdate", false, "")
+	flag.BoolVar(&noSelfUpdate, "no-selfupdate", false, "")
+	flag.IntVar(&migrationNum, "migration", 0, "")
+	flag.StringVar(&fsRoot, "root", "", "")
+
+	flag.Parse()
+	// this is only to keep the flag parsing logic simple and avoid double
+	// negations/negatives below
+	selfUpdate := !noSelfUpdate
+
+	if verbose {
+		connect.EnableDebug()
+	}
+
+	if !quiet {
+		connect.QuietOut.SetOutput(os.Stdout)
+	}
+
+	connect.CFG.Load()
+
+	// pass root to connect config
+	if fsRoot != "" {
+		connect.CFG.FsRoot = fsRoot
+		// if we update a chroot system, we cannot create snapshots of it
+		noSnapshots = true
+	}
+
+	if !isSnapperConfigured() {
+		noSnapshots = true
+		connect.Debug.Println("Snapper not configured")
```

Comment:
maybe we could use shortcuts for these loggers? Something like
`var Debug = connect.Debug`

Reply:
done. for now with `Debug` and `QuietOut` but these could be unexported as they won't be used from outside.